csharp/src/Ice/ACM.cs

```diff
@@ -105,204 +162,125 @@ internal void Destroy()
-                }
-                else
+                if (_reapedConnections.Count == 0)
                 {
-                    _changes.Add(new Change(connection, false));
+                    return Enumerable.Empty<Connection>();
                 }
+                List<Connection> connections = _reapedConnections;
+                _reapedConnections = new List<Connection>();
+                return connections;
             }
         }
 
-        public void Remove(Connection connection)
-        {
-            if (_config.Timeout == 0)
-            {
-                return;
-            }
-
-            lock (_mutex)
-            {
-                Debug.Assert(_communicator != null);
-                _changes.Add(new Change(connection, true));
-            }
-        }
-
-        public void Reap(Connection connection)
-        {
-            lock (_mutex)
-            {
-                _reapedConnections.Add(connection);
-            }
-        }
-
-        public IACMMonitor Acm(int? timeout, ACMClose? c, ACMHeartbeat? h)
-        {
-            Debug.Assert(_communicator != null);
-
-            ACMConfig config = _config.Clone();
-            if (timeout.HasValue)
-            {
-                config.Timeout = timeout.Value * 1000; // To milliseconds
-            }
-            if (c.HasValue)
-            {
-                config.Close = c.Value;
-            }
-            if (h.HasValue)
-            {
-                config.Heartbeat = h.Value;
-            }
-            return new ConnectionACMMonitor(this, _communicator.Timer(), config);
-        }
-
-        public ACM GetACM()
-        {
-            return new ACM
-            {
-                Timeout = _config.Timeout / 1000,
-                Close = _config.Close,
-                Heartbeat = _config.Heartbeat
-            };
-        }
-
-        internal IEnumerable<Connection> SwapReapedConnections()
+        internal void HandleException(Exception ex)
         {
             lock (_mutex)
             {
-                if (_reapedConnections.Count == 0)
+                if (_communicator == null)
                 {
-                    return Enumerable.Empty<Connection>();
+                    return;
                 }
-                List<Connection> connections = _reapedConnections;
-                _reapedConnections = new List<Connection>();
-                return connections;
+                _communicator.Logger.Error("exception in connection monitor:\n" + ex);
             }
         }
 
-        public void RunTimerTask()
+        private void RunTimerTask(object? state)
         {
             lock (_mutex)
             {
                 if (_communicator == null)
                 {
                     _connections.Clear();
-                    System.Threading.Monitor.PulseAll(_mutex);
+                    Monitor.PulseAll(_mutex);
                     return;
                 }
```

Comment:
This could be removed, _communicator should not longer be set to null by Destroy.

Reply:
Fixed